### PR TITLE
Add a method to get session secret for onion packet to KeysInterface

### PR DESCRIPTION
**Pull request by yuntai**
per 
https://github.com/rust-bitcoin/rust-lightning/pull/243#discussion_r236018505

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Nice, looks good, need to get implemented in full_stack_target, though.

**Comment by yuntai** [comment]
oops. fixed.

File: src/chain/keysinterface.rs [hunk 1]
```diff
@@ -235,4 +242,9 @@ impl KeysInterface for KeysManager {
 		sha.result(&mut seed);
 		ChannelKeys::new_from_seed(&seed)
 	}
+
+	fn get_session_key(&self) -> SecretKey {
+		let child_ix = self.session_child_index.fetch_add(1, Ordering::AcqRel);
+		self.session_master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(child_ix as u32)).expect("Your RNG is busted").secret_key
```
Comment (TheBlueMatt) [inline src/chain/keysinterface.rs]: Oops, sorry, this isnt sufficient as it'll wrap back to 0 on restart and we'll re-use session keys. Can you hash in the timestamp like we do for channel keys? I'm not too worried about needing extra rng output, here, though.

File: fuzz/fuzz_targets/full_stack_target.rs [hunk 2]
```diff
@@ -265,6 +265,10 @@ impl KeysInterface for KeyProvider {
 			}
 		}
 	}
+
+	fn get_session_key(&self) -> SecretKey {
+		SecretKey::from_slice(&Secp256k1::without_caps(), &[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]).unwrap()
```
Comment (TheBlueMatt) [inline fuzz/fuzz_targets/full_stack_target.rs]: Oops, please avoid changing the keys used in things in the fuzz targets. You can just take https://github.com/TheBlueMatt/rust-lightning/commit/5a829777a67bd5d3c09ae4c5e6ef2cf81c88c6af, I think.

**Comment by yuntai** [comment]
updated. not sure ...expect("RNG is busted") is correct things to print out

**Comment by TheBlueMatt** [comment]
Looks good, sans the compile error that broke travis.

**Pull request merged**